control/plugin/client/native.go

```diff
@@ -92,8 +94,47 @@ func (p *PluginNativeClient) Kill(reason string) error {
 	return err
 }
 
-func (p *PluginNativeClient) Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error {
-	args := plugin.PublishArgs{ContentType: contentType, Content: content, Config: config}
+func encodeMetrics(metrics []core.Metric) []byte {
+	mts := make([]plugin.MetricType, len(metrics))
+	for i, m := range metrics {
+		mts[i] = plugin.MetricType{
+			Namespace_:          m.Namespace(),
+			Tags_:               m.Tags(),
+			Timestamp_:          m.Timestamp(),
+			Version_:            m.Version(),
+			Config_:             m.Config(),
+			LastAdvertisedTime_: m.LastAdvertisedTime(),
```

Comment:
There is a method ToCoreMetric which should be used or removed but more importantly in it we are checking if LastAdvertisedTime.Sec was `-62135596800` and if so we would set LastAdvertisedTime to time.Now().  This still needs to happen otherwise in the case were we try to gob.Encode the metrics the issue that caused us to put in this check will resurface (#1118).

We should also perform the same check on the Timestamp field.